README.md

```diff
@@ -5,96 +5,144 @@
+  - [Local development workflow using `yarn link`](#local-development-workflow-using-yarn-link)
+- [Browser support](#browser-support)
+- [Contributing](#contributing)
+
+## Usage guide
+
+This guide contains a few [usage examples](#usage-examples) and links to [sample applications](#sample-applications) which demonstrate how to use the Okta Sign-In Widget in various scenarios.
+
+Based upon your needs, you may want to use the [default Okta-hosted signin page](#okta-hosted-signin-page-default) or a [custom domain][] with a [customizable Okta-hosted signin page](#okta-hosted-signin-page-customizable). For maximum customization and a seamless experience, you can [embed the widget](#embedded-self-hosted) directly into your application.
+
+Okta also provides several higher-level [SDKs](#sdks) which can help integrate Okta features into your application.
+
+### Okta-hosted signin page (default)
+
+Okta provides a signin page, available at your [organization][]'s URL which allows the user to complete the entire authorization flow, start a SSO (Single Sign-On) session, and set the Okta [session cookie][] in the web browser. By default, signing in on this page will redirect the user to the Okta user dashboard.
+
+The default Okta-hosted signin page can also be used to authenticate a user in a web or mobile app. Your app can [redirect to a sigin-in page][] to perform the [authentication][] flow after which Okta will redirect back to an app [callback][] which either contains [OAuth][] tokens (in the case of the [implicit flow][]) or a code which can be used to obtain [OAuth][] tokens (in the case of [PKCE][] and the [authorization code flow][]). Okta provides [SDKs](#sdks) in many languages to help construct the redirect URL and handle the login [callback][] as part of the [hosted flow][].
+
+Okta provides several complete [sample applications](#sample-applications) which demonstrate how to use the Okta [hosted flow][].
+
+### Okta-hosted signin page (customizable)
+
+The default Okta hosted signin page can be customized so that it is available under a [custom domain][] which is a subdomain of your company's top-level domain. Although the page is hosted by Okta, you are able to [customize the template][] of this page in many powerful ways. See the [gallery][] for some examples of customized widgets.
+
+As far as your app is concerned, the customized widget behaves the same as the default Okta-hosted widget and you can use the same [hosted flow][].
+
+### Embedded (self-hosted)
+
+For a completely seamless experience, the Sign-In Widget can be embedded directly into your web application. This allows for the highest level of customization. Using an embedded widget, client-side web and native apps using [PKCE][] or [implicit flow][] can avoid the round-trip redirect of the [hosted flow][]. An embedded widget can perform the complete [OIDC][] flow and return [OAuth][] tokens directly to the application. See [showSignInToGetTokens](#showsignintogettokens).
```

Comment:
This is a good top-level overview of the 3 main widget hosting modes. It would be great if we included some of the core terminology as well such as "Federated Client" vs "Direct Authentication Client" and "ID Site", etc. It is important to define all the primitives and then also explain what is NOT supported currently as well. I think we also want to surface the first principles (all redirects are PKCE-bound, no tokens over a URL, etc).

Reply:
Back up on line 35 is a **very** brief overview of what the widget can do. But still missing from this document is a section explaining how the widget fits into a larger design. Where does the widget fit into a federated / direct / IDP scenario? Maybe you have or know of some resources that could help fill out this section? 

So far I'm working with these links:

Authentication (overview)
https://developer.okta.com/docs/concepts/authentication/

Federated
https://developer.okta.com/books/api-security/authn/federated/

Direct Auth: 
https://developer.okta.com/docs/reference/api/authn/

IDP
https://developer.okta.com/docs/concepts/identity-providers/

Reply:
Added a JIRA to follow up and add more detail on federated/IDP scenarios: https://oktainc.atlassian.net/browse/OKTA-357671